src/states/node.rs

```diff
@@ -1722,13 +1710,37 @@ impl Node {
             return Err(RoutingError::UnknownConnection(pub_id));
         };
 
+        let ip = if let Ok(ip) = self.crust_service.get_peer_ip_addr(&pub_id) {
+            ip
+        } else {
+            debug!(
+                "{:?} Can't get IP address of bootstrapper {:?}.",
+                self,
+                pub_id
+            );
+            self.disconnect_peer(&pub_id, None);
+            return Err(RoutingError::UnknownConnection(pub_id));
+        };
+
+        if peer_kind == CrustUser::Client && !self.peer_mgr.can_accept_client(ip) {
```

Comment:
is that the above `fetch ip` operation shall only happen for `peer_kind == CrustUser::Client` ?
and, when disconnecting peer (reject or fail to fetch ip), it shall be added into dropped_clients ?